### PR TITLE
Add DepthSense SDK grabber and viewer

### PR DESCRIPTION
**Pull request by taketwo**
This pull request adds a new grabber for DepthSense devices (e.g. [Creative Senz3D](http://de.creative.com/p/web-cameras/creative-senz3d),
[SoftKinetic DS325](http://www.softkinetic.com/language/fr-BE/Products/DepthSenseCameras)) that supports both Linux and Windows versions of the SDK. The grabber has been residing in [my private repository](https://github.com/taketwo/ds) for quite some time already, but now it's time to merge it upstream.

At the moment there is one nasty CMake problem that I am not able to figure out. An advice from an expert (@jspricke ?) would be much appreciated. The DepthSense SDK on Linux is shipped with it's own `libusb-1.0.so` file, which ends up in the installation directory `/opt/softkinetic/DepthSenseSDK/lib/libusb-1.0.so.0`. In PCL we are also using this library, but the system version of course. While configuring, CMake properly finds the system version:

```
Found LIBUSB_1: /usr/lib/x86_64-linux-gnu/libusb-1.0.so
```

However, further on I get tons of warnings like this:

```
CMake Warning at cmake/pcl_targets.cmake:327 (add_executable):
  Cannot generate a safe runtime search path for target pcl_openni_image
  because files in some directories may conflict with libraries in implicit
  directories:

    runtime library [libusb-1.0.so.0] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /opt/softkinetic/DepthSenseSDK/lib

  Some of these libraries may not be found correctly.
```

Any idea how to tell CMake to use system version always?

### CODE REVIEW

File: io/include/pcl/io/depth_sense/depth_sense_device_manager.h [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+            if (!instance)
+            {
+              boost::mutex::scoped_lock lock (mutex_);
+              if (!instance)
+                instance.reset (new DepthSenseDeviceManager);
+            }
+            return (instance);
+          }
+
+          /** Get the number of connected DepthSense devices. */
```
Comment (VictorLamoine) [inline io/include/pcl/io/depth_sense/depth_sense_device_manager.h]: I don't understand why I don't get doxygen warning about this but the documentation is incomplete; you should use `@param`, `@return` for a clean documentation.

There are lots of undocumented stuff in here.

Comment (taketwo) [inline io/include/pcl/io/depth_sense/depth_sense_device_manager.h]: I'm not a big fan of writing documentation for the sake of documentation. For example, in this particular function the `\brief` description gives all the information, there is no need to say exactly the same once again in `\return`.

**Comment by taketwo** [comment]
Okay, I solved the CMake problem by simply deleting the `libusb-1.0.so` shipped with the SDK. The programs shipped with the SDK continue to work when this dependency is resolved to the system library, and so does the grabber. I will mention this (as well as some other installation issues) in a tutorial which will be submitted later.

I tested this pull request with both Linux and Windows, works fine on both.

**Comment by taketwo** [comment]
One related question is what to do with `PXCGrabber`? This is a Windows-only grabber for the _same_ family of devices, which is based on [Intel® Perceptual Computing SDK](https://software.intel.com/en-us/perceptual-computing-sdk). That SDK is obsolete and was discontinued in August 2014. It is not available for download anymore.

I propose to remove it from PCL completely, leaving only empty include file with `error` macro (pointing to the new grabber).

**Comment by taketwo** [comment]
Updated the pull request to match the new approach to `WITH_*` options in _CMake_ scripts. Also adopted Piml pattern to hide the dependency on DSSDK from the users of `DepthSenseGrabber`.

**Comment by taketwo** [comment]
@jspricke ?

**Comment by jspricke** [comment]
Will merge once the last comment is resolved. +1 for removing the old driver as well.

**Comment by taketwo** [comment]
Thanks! I'll submit a PR that removes the old driver later.

**Comment by jspricke** [comment]
Thank you!

**Pull request merged**